lib/krane/kubernetes_resource/daemon_set.rb

```diff
@@ -78,9 +78,10 @@ def rollout_data
 
     def parent_of_pod?(pod_data)
       return false unless pod_data.dig("metadata", "ownerReferences")
+      template_generation = @instance_data["spec"]["templateGeneration"] ||
+        @instance_data["metadata"]["annotations"]["deprecated.daemonset.template.generation"]
```

Comment:
Let's open an issue to find a long term solution here.

Reply:
Done: https://github.com/Shopify/kubernetes-deploy/issues/627

Reply:
Should we actually be using `dig` in case annotations don't exist?

Reply:
Good idea! Implemented 👍